app/services/search_algorithms.rb

```diff
@@ -0,0 +1,88 @@
+class SearchAlgorithms
```

Comment:
I think it's nice to have all the raw SQL in once place, separate from the service.